index.js

```diff
@@ -8,14 +8,14 @@ var crypto = require('crypto');
  * Sign the given `val` with `secret`.
  *
  * @param {String} val
- * @param {String} secret
+ * @param {String|NodeJS.ArrayBufferView|crypto.KeyObject} secret
  * @return {String}
  * @api private
  */
 
 exports.sign = function(val, secret){
-  if ('string' != typeof val) throw new TypeError("Cookie value must be provided as a string.");
-  if ('string' != typeof secret) throw new TypeError("Secret string must be provided.");
+  if ('string' !== typeof val) throw new TypeError("Cookie value must be provided as a string.");
+  if (null === secret) throw new TypeError("Secret key must be provided.");
```

Comment:
However, this is incorrect!

Because `null === secret` matches `null` and only `null`. 

But `null == secret` matches both if secret is `null` and **also** if secret is not passed at all (`undefined`) or comes from a missing property, etc. So we need to keep the non-strict (`==`) equality here.

```suggestion
  if (null == secret) throw new TypeError("Secret key must be provided.");
```